### PR TITLE
[FIX] specify dtype when loading gifti

### LINKED ISSUE #3638: [BUG] `ValueError` when loading fsaverage surface with `numpy` 1.24.x installed
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Operating system

- [ ] Linux
- [ ] Mac
- [X] Windows

### Operating system version

- Windows 10

### Python version

- [ ] 3.11
- [X] 3.10
- [ ] 3.9
- [ ] 3.8
- [ ] 3.7

### nilearn version

0.10.0

### Expected behavior

Load the data without errors

### Current behavior & error messages

This is what I got:

```bash
File "c:\users\dalth\.spyder-py3\temp.py", line 8, in <module>
    brain_vertices, brain_faces = surface.load_surf_data(fsaverage['pial_left'])

  File "C:\Users\dalth\anaconda3\envs\napari-dev\lib\site-packages\nilearn\surface\surface.py", line 733, in load_surf_data
    data_part = _gifti_img_to_data(gii)

  File "C:\Users\dalth\anaconda3\envs\napari-dev\lib\site-packages\nilearn\surface\surface.py", line 684, in _gifti_img_to_data
    return np.asarray([arr.data for arr in gifti_img.darrays]).T.squeeze()

ValueError: setting an array element with a sequence. The requested array has an inhomogeneous shape after 1 dimensions. The detected shape was (2,) + inhomogeneous part.
```

### Steps and code to reproduce bug

Hi, the example code to reproduce the issue:

```python
from nilearn import datasets, surface

# Fetch datasets - this will download dataset if datasets are not found
nki_dataset = datasets.fetch_surf_nki_enhanced(n_subjects=1)
fsaverage = datasets.fetch_surf_fsaverage()

# Load surface data and resting state time series from nilearn
brain_vertices, brain_faces = surface.load_surf_data(fsaverage['pial_left'])
```

Running the code above with `numpy` 1.24.x installed ends up in a `ValueError`:

```
ValueError: setting an array element with a sequence. The requested array has an inhomogeneous shape after 1 dimensions. The detected shape was (2,) + inhomogeneous part.
```

While using `numpy` 1.23.x shows a `VisibleDeprecationWarning`:

```
\lib\site-packages\nilearn\surface\surface.py:684: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.
  return np.asarray([arr.data for arr in gifti_img.darrays]).T.squeeze()
```

Seems like what is missing to be compatible with `numpy` 1.24.x is adding `dtype=object` to the `np.asarray` call at https://github.com/nilearn/nilearn/blob/2b69b29cbdbfce4fa94bf4f07fc15b0c44bc6985/nilearn/surface/surface.py#L684 ?

### PR DESCRIPTION
**Pull request by Remi-Gau**
Closes #3638

Changes proposed in this pull request:

- specify dtype when loading data from gifti in `nilearn.surface._gifti_img_to_data`

As far as I can tell this should fail with the oldest numpy version we support: https://numpy.org/doc/1.19/reference/generated/numpy.asarray.html?highlight=asarray#numpy.asarray

### CODE REVIEW

**Review by bthirion — APPROVED** [review]
LGTM, thx.

File: nilearn/surface/tests/test_surface.py [hunk 1]
```diff
@@ -66,6 +64,19 @@ def data(self):
         return self._data
 
 
+def test_load_surf_data_numpy_gt_1pt23():
+    """Test loading fsaverage surface 
+    
+    Threw an error with numpy >=1.24.x
+    but only a deprecaton warning with numpy <1.24.x.
```
Comment (ymzayek) [inline nilearn/surface/tests/test_surface.py:71]:
```suggestion
    but only a deprecaton warning with numpy <1.23.x.
```

**Review by ymzayek — APPROVED** [review]
LGTM thanks @Remi-Gau

**Comment by ymzayek** [comment]
@Remi-Gau for this failure https://github.com/nilearn/nilearn/pull/3644#issuecomment-1495553708, can't we just return a single data array as is for giftis containing one data array? That will just return that array with its data type without setting `dtype=object` like we would have to do in the case of 2 data arrays of non matching shapes. In other words we could add:

```python
    if len(gifti_img.darrays) == 1:
        return np.asarray([gifti_img.darrays[0].data]).T.squeeze()
```

in `def _gifti_img_to_data` before the final return

**Comment by Remi-Gau** [comment]
> Thx @Remi-Gau !

Thanks go to you: you cracked this one, not I. 🤗

**Pull request merged**